### ISSUE TITLE
Why is duktape.c distributed with '#line' cpp entries?

### ISSUE THREAD

**Issue by glima** [svaarala/duktape#362]
If it's purposed to be embedded in other projects at arbitrary locations? Generating debug packages spanning all the code for such derivative projects is then made clumsy: either you have to patch the fetched duktape release, removing those lines, or you get

`cpio: <arbitrary_path>/duk_util_hashbytes.c: Cannot stat: No such file or directory`

messages when rpmbuild is trying to generate debug packages.

**Comment by svaarala** [comment]
One reason is to allow error messages to use `__LINE__` and still refer to the original source lines. For example:

```
((o) Duktape 1.3.99 (v1.3.0-44-ga6c3826)
duk> decodeURIComponent('%ff')
URIError: invalid input
	duk_bi_global.c:343
	decodeURIComponent  native strict preventsyield
	global input:1 preventsyield
```

Without the #line directives, that `duk_bi_global.c:343` would refer to the combined source and be less informative. It would be trivial to remove the #line directives, but it'd have some downsides like this.

For debugging one can provide the uncombined sources in e.g. gdb search path, so that you're actually debugging with the uncombined sources rather than the large single source.

Another option would be to build from the separate source files.

**Comment by svaarala** [comment]
It would also be straightforward to create two different combined source files - one with #line directives and another without. The downside with that would be that it'd need to be explained and it gets somewhat technical so I'd prefer having either or.

Anyway, we can remove the #line directives if the benefits are worth it. I remember I didn't have them originally and some issue prompted me to add them, probably that traceback issue.

**Comment by fatcerberus** [comment]
The #line directives are useful for reporting compiler errors and such... This way they can be traced back to the proper source file and line.  Otherwise the information is meaningless.

**Comment by svaarala** [comment]
In this case there are two relevant source files though: the combined source and the individual source.

**Comment by glima** [comment]
Yep, the reason for the individual source having this kind of info is void to me.

**Comment by fatcerberus** [comment]
As mentioned in #363, removing the #line directives breaks debugging under MSVC because the combined source is >64k lines.  So no matter what is done here, something is going to be broken, unfortunately. :(

**Comment by svaarala** [comment]
Ah, I was more or less clear that both are needed - but I'm not sure which should be the default.

**Comment by svaarala** [comment]
This was fixed in 1.4.0 so closing the issue.
